src/main/java/teammates/ui/controller/InstructorFeedbackResultsPageData.java

```diff
@@ -1172,9 +1172,11 @@ private void buildTableColumnHeaderForRecipientQuestionGiverView(List<ElementTag
             responseRows.add(responseRow);
         }
         
-        responseRows.addAll(getRemainingMissingResponseRows(question, possibleGiversWithoutResponses, 
-                                                            possibleReceiversWithoutResponsesForGiver, 
-                                                            prevGiver, viewType));
+        if (!responses.isEmpty()) {
+            responseRows.addAll(getRemainingMissingResponseRows(question, possibleGiversWithoutResponses, 
+                                                                possibleReceiversWithoutResponsesForGiver, 
+                                                                prevGiver, viewType));
+        }
```

Comment:
This applies to sessions with 0 responses. As soon as it has even one response, it will become as slow as before :-p